amgut/db/patches/0029.sql

```diff
@@ -0,0 +1,155 @@
+-- October 12, 2015
+-- Add tables needed to track plate maps and create prep templates
+--Need to wipe out old, bad data in the plate table
+DROP TABLE barcodes.plate_barcode CASCADE;
+DELETE FROM barcodes.plate;
+
+ALTER TABLE barcodes.barcode ALTER COLUMN barcode TYPE varchar;
+ALTER TABLE barcodes.project ADD description varchar  NOT NULL DEFAULT '';
+
+CREATE TABLE barcodes.lot ( 
+	lot_id               bigserial  NOT NULL,
+	lot_type             varchar  NOT NULL,
+	lot                  varchar  NOT NULL,
+	date_added           date DEFAULT current_date NOT NULL,
+	CONSTRAINT pk_lots PRIMARY KEY ( lot_id )
+ );
+
+CREATE TABLE barcodes.robots ( 
+	robot_id             bigserial  NOT NULL,
+	robot_type           varchar  NOT NULL,
+	robot_name           varchar(100)  NOT NULL,
+	date_added           date DEFAULT current_date NOT NULL,
+	CONSTRAINT pk_robots PRIMARY KEY ( robot_id )
+ );
+
+CREATE TABLE barcodes.person ( 
+	person_id            bigint  NOT NULL,
+	name                 varchar(100)  NOT NULL,
+	email                varchar  NOT NULL,
+	date_added           date DEFAULT current_date NOT NULL,
+	CONSTRAINT pk_person PRIMARY KEY ( person_id )
+ );
+
+CREATE TABLE barcodes.protocol ( 
+	protocol_id          bigserial  NOT NULL,
+	protocol_name        varchar(100)  NOT NULL,
+	template_dna         integer  NOT NULL,
+	platform             varchar  NOT NULL,
+	pcr_primers          varchar  NOT NULL,
+	instrument_model     varchar  NOT NULL,
+	linker               varchar  NOT NULL,
+	sequencing_method    varchar  NOT NULL,
+	target_gene          varchar  NOT NULL,
+	target_subfragment   varchar  NOT NULL,
+	CONSTRAINT pk_protocol PRIMARY KEY ( protocol_id )
+ );
+COMMENT ON COLUMN barcodes.protocol.template_dna IS 'Template DNA added, in uL';
+
+CREATE TABLE barcodes.pcr_plate_run ( 
+	run_id               bigint  NOT NULL,
+	barcode              varchar  NOT NULL,
+	nickname             varchar(100)  NOT NULL,
```

Comment:
multiple pcr plates from a single extraction plate, so need a way to track them. They are not kept, so they don't get a barcode, but we still need to differentiate them.